fastapi/readme/USAGE.rst

```diff
@@ -0,0 +1,1157 @@
+  able to perform an appropriate action specific to this record (for example,
+  the processing of a payment is different for each payment acquirer). In this
+  case, you must provide a specific attribute that allows both the client and
+  odoo to identify the record. The field 'provider' on a payment acquirer allows
+  you to identify a specific record in odoo. This kind of approach
+  allows both the client and odoo to identify the record without having to rely
+  on the id of the record. (This will ensure that the client will not break
+  if the id of the record is changed in odoo for example when tests are run
+  on an other database).
+
+* Always use the same name for the same kind of object. For example, if you
+  have a field 'lines' that contains a list of sale order lines, you must
+  use the same name for the same kind of object in all the other json documents.
+
+* Manage relations between objects in your json documents the same way.
+  By default, you should return the id of the related object in the json document.
+  But this is not always possible or convenient, so you can also return the
+  related object in the json document. The main advantage of returning the id
+  of the related object is that it allows you to avoid the `n+1 problem
+  <https://restfulapi.net/rest-api-n-1-problem/>`_ . The
+  main advantage of returning the related object in the json document is that
+  it allows you to avoid an extra call to retrieve the related object.
+  By keeping in mind the pros and cons of each approach, you can choose the
+  best one for your use case. Once it's done, you must be consistent in the
+  way you manage the relations of the same object.
+
+* It's not always a good idea to name your fields into your json documents
+  with the same name as the fields of the corresponding odoo model. For example,
+  in your document representing a sale order, you must not use the name 'order_line'
+  for the field that contains the list of sale order lines. The name 'order_line'
+  in addition to being confusing and not consistent with the best practices, is
+  not auto-descriptive. The name 'lines' is much better.
+
+* Keep a defensive programming approach. If you provide a route handler that
+  returns a list of records, you must ensure that the computation of the list
+  is not too long or will not drain your server resources. For example,
+  for search route handlers, you must ensure that the search is limited to
+  a reasonable number of records by default.
+
+* As a corollary of the previous point, a search handler must always use the
+  pagination mechanism with a reasonable default page size. The result list
+  must be enclosed in a json document that contains the total number of records
+  and the list of records.
+
+* Use plural for the name of a service. For example, if you provide a service
+  that allows you to manage the sale orders, you must use the name 'sale_orders'
+  and not 'sale_order'.
```

Comment:
Mention the concept of collection of resources which must be named with a plural (`/sale_orders/{id}`?